pkg/document/document.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package document
+
+import "encoding/json"
+
+const jsonldID = "id"
+
+// Document defines generic document data structure
+type Document map[string]interface{}
+
+// ID is document identifier
+func (doc *Document) ID() string {
+	return stringEntry((*doc)[jsonldID])
+}
+
+// FromBytes creates an instance of Document by reading a JSON document from bytes
+func FromBytes(data []byte) (Document, error) {
+
+	doc := make(Document)
+	err := json.Unmarshal(data, &doc)
+	if err != nil {
+		return nil, err
+	}
+
+	return doc, nil
+}
+
+// Bytes returns byte representation of did document
+func (doc *Document) Bytes() []byte {
+	s, err := json.Marshal(doc)
+	if err != nil {
+		return []byte("<ERROR marshalling Document>")
```

Comment:
Where does this error string appear?

Reply:
It is very unlikely it will appear since it is hard to produce this error, see TestMarshalError.

Reply:
Fixed